hera_qm/xrfi.py

```diff
@@ -607,11 +685,215 @@ def xrfi_h1c_pipe(uv, Kt=8, Kf=8, sig_init=6., sig_adj=2., px_threshold=0.2,
+    """xrfi excision pipeline used for H1C IDR2.2. Uses detrending and watershed algorithms above.
+    Args:
+        ocalfits_file (str): Omnical calfits file to use to flag on gains and
+            chisquared values.
+        acalfits_file (str): Abscal calfits file to use to flag on gains and
+            chisquared values.
+        model_file (str): Model visibility file to flag on.
+        data_file (str): Raw visibility data file to flag.
+        history (str): History string to include in files
+        init_metrics_ext (str, optional): Extension to be appended to input file name
+            for initial metric object. Default is "init_xrfi_metrics.h5".
+        init_flags_ext (str, optional): Extension to be appended to input file name
+            for initial flag object. Default is "init_flags.h5".
+        final_metrics_ext (str, optional): Extension to be appended to input file name
+            for final metric object. Default is "final_xrfi_metrics.h5".
+        final_flags_ext (str, optional): Extension to be appended to input file name
+            for final flag object. Default is "final_flags.h5".
+        xrfi_path (str, optional): Path to save xrfi files to. Default is same
+            directory as data_file.
+        kt_size (int, optional): Size of kernel in time dimension for detrend in
+            xrfi algorithm. Default is 8.
+        kf_size (int, optional): Size of kernel in frequency dimension for detrend
+            in xrfi algorithm. Default is 8.
+        sig_init (float, optional): Starting number of sigmas to flag on. Default is 5.
+        sig_adj (float, optional): Number of sigmas to flag on for data adjacent
+            to a flag. Default is 2.0.
+        freq_threshold (float, optional): Fraction of times required to trigger
+            broadcast across times (single freq). Default is 0.35.
+        time_threshold (float, optional): Fraction of channels required to trigger
+            broadcast across frequency (single time). Default is 0.5.
+        ex_ants (str, optional): Comma-separated list of antennas to exclude.
+            Flags of visibilities formed with these antennas will be set to True.
+        metrics_file (str, optional): Metrics file that contains a list of excluded
+            antennas. Flags of visibilities formed with these antennas will be set to True.
+        cal_ext (str, optional): Extension to replace penultimate extension in
+            calfits file for output calibration including flags. Defaults is "flagged_abs".
+            For example, an input_cal of "foo.goo.calfits" would result in
+            "foo.flagged_abs.calfits".
+    Returns:
+        None
+    """
+    history = 'Flagging command: "' + history + '", Using ' + hera_qm_version_str
+    dirname = resolve_xrfi_path(xrfi_path, data_file)
+    xants = process_ex_ants(ex_ants=ex_ants, metrics_file=metrics_file)
+
+    # Initial run on cal data products
+    alg = 'detrend_medfilt'
```

Comment:
I don't see the need to make this its own variable unless you want it to be a kwarg. I think it makes the calls to xrfi_pipe less readable.

Reply:
Hmm. I actually see the variable as useful in case we decide to change it, there's only one point to change. Nevertheless I took it out of the code for now. Maybe @plaplant can be the deciding vote when he reviews.